src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -0,0 +1,169 @@
+      // Operational part of while-loop
+      mask.add(this.coordToDataArrayIndices(x, y, imgData.width)[0]);
+
+      // Loop part of while-loop
+
+      // Get coords of adjacent pixels
+      let leftPixelCoord: Array<number> = [x - 1, y];
+      let rightPixelCoord: Array<number> = [x + 1, y];
+      let upPixelCoord: Array<number> = [x, y - 1];
+      let downPixelCoord: Array<number> = [x, y + 1];
+      // Add coords of adjacent pixels to the heap
+      let isStillMask: boolean = this.getIsMask(originalPixel, imgData, leftPixelCoord, tolerance, visited);
+      // Check left pixel
+      if (isStillMask) {
+        visit.push(leftPixelCoord);
+      }
+      isStillMask = this.getIsMask(originalPixel, imgData, rightPixelCoord, tolerance, visited);
+      // Check right pixel
+      if (isStillMask) {
+        visit.push(rightPixelCoord);
+      }
+      isStillMask = this.getIsMask(originalPixel, imgData, upPixelCoord, tolerance, visited);
+      // Check up pixel
+      if (isStillMask) {
+        visit.push(upPixelCoord);
+      }
+      isStillMask = this.getIsMask(originalPixel, imgData, downPixelCoord, tolerance, visited);
+      // Check down pixel
+      if (isStillMask) {
+        visit.push(downPixelCoord);
+      }
+    }  // end of while loop
+
+    return mask;
+  }
+
+  /**First check that @pixelCoord is within range of the image bounds.
+   * Then, judge current pixel's RGBA against original pixel's RGBA to
+   * see if it can still be part of the mask (using tolerance criteria).
+   */
+  getIsMask(originalPixel: Array<number>, imgData: ImageData,
+    pixelCoord: Array<number>, tolerance: number,
+    visited: Set<number>): boolean {
+    let curX: number = pixelCoord[0];
+    let curY: number = pixelCoord[1];
+    let imgWidth: number = imgData.width;
+
+    // Preface; check if pixel is valid (indexing errs and repeat values)
+    let isValid: boolean = this.getIsValid(imgWidth, imgData.height, curX, curY, visited);
+    if (!isValid) {
+      // Automatically not in mask b/c failed vailidity test
+      return false;
+    }
+
+    visited.add(this.coordToDataArrayIndices(curX, curY, imgWidth)[0]);
+
+    // Get array of attributes of current pixel
+    let curPixel: Array<number> = this.dataArrayToRGBA(imgData, curX, curY);
+
+    // All attributes of the pixel (R,G,B, and A) must be within tolerance level
+    for (let i = 0; i < 4; i++) {
+      let upperTolerance: boolean = curPixel[i] > originalPixel[i] + tolerance;
+      let lowerTolerance: boolean = curPixel[i] < originalPixel[i] - tolerance;
+      if (upperTolerance || lowerTolerance) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  // Check if @pixelCoord is in bounds and makes sure it's not a repeat coord
+  getIsValid(imgWidth: number, imgHeight: number, curX: number, curY: number,
+    visited: Set<number>): boolean {
+    if (this.isInBounds(imgWidth, imgHeight, curX, curY) &&
+      this.notVisited(imgWidth, curX, curY, visited)) {
```

Comment:
nit: line wrapping continuation should be +4 spaces

https://google.github.io/styleguide/jsguide.html#formatting-indent